### ISSUE TITLE
Reload segfaults if routes are defined

### ISSUE THREAD

**Issue by psuet** [acassen/keepalived#2360]
**Describe the bug**
When defining static default routes for IPv4 and IPv6, keepalived reproducibly segfaults on reload, but not on reboot.

**To Reproduce**
Reload the service

**Expected behavior**
Reloading should succeed without coredump

**Keepalived version**
<pre>Keepalived v2.2.7 (01/16,2022)

Copyright(C) 2001-2022 Alexandre Cassen, &lt;acassen@gmail.com&gt;

Built with kernel headers for Linux 5.19.11
Running on Linux 6.1.0-13-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.55-1 (2023-09-29)
Distro: Debian GNU/Linux 12 (bookworm)

configure options: --build=x86_64-linux-gnu --prefix=/usr --includedir=${prefix}/include --mandir=${prefix}/share/man --infodir=${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --disable-option-checking --disable-silent-rules --libdir=${prefix}/lib/x86_64-linux-gnu --runstatedir=/run --disable-maintainer-mode --disable-dependency-tracking --enable-snmp --enable-sha1 --enable-snmp-rfcv2 --enable-snmp-rfcv3 --enable-dbus --enable-json --enable-bfd --enable-regex --with-init=systemd build_alias=x86_64-linux-gnu CFLAGS=-g -O2 -ffile-prefix-map=/build/keepalived-m8ENAG/keepalived-2.2.7=. -fstack-protector-strong -Wformat -Werror=format-security LDFLAGS=-Wl,-z,relro CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2

Config options:  NFTABLES LVS REGEX VRRP VRRP_AUTH VRRP_VMAC JSON BFD OLD_CHKSUM_COMPAT SNMP_V3_FOR_V2 SNMP_VRRP SNMP_CHECKER SNMP_RFCV2 SNMP_RFCV3 DBUS INIT=systemd SYSTEMD_NOTIFY

System options:  VSYSLOG MEMFD_CREATE IPV6_MULTICAST_ALL IPV4_DEVCONF LIBNL3 RTA_ENCAP RTA_EXPIRES RTA_NEWDST RTA_PREF FRA_SUPPRESS_PREFIXLEN FRA_SUPPRESS_IFGROUP FRA_TUN_ID RTAX_CC_ALGO RTAX_QUICKACK RTEXT_FILTER_SKIP_STATS FRA_L3MDEV FRA_UID_RANGE RTAX_FASTOPEN_NO_COOKIE RTA_VIA FRA_PROTOCOL FRA_IP_PROTO FRA_SPORT_RANGE FRA_DPORT_RANGE RTA_TTL_PROPAGATE IFA_FLAGS LWTUNNEL_ENCAP_MPLS LWTUNNEL_ENCAP_ILA NET_LINUX_IF_H_COLLISION LIBIPVS_NETLINK IPVS_DEST_ATTR_ADDR_FAMILY IPVS_SYNCD_ATTRIBUTES IPVS_64BIT_STATS IPVS_TUN_TYPE IPVS_TUN_CSUM IPVS_TUN_GRE VRRP_IPVLAN IFLA_LINK_NETNSID GLOB_BRACE GLOB_ALTDIRFUNC INET6_ADDR_GEN_MODE VRF SO_MARK
</pre>

**Distro (please complete the following information):**
 - Name: Debian
 - Version: 12
 - Architecture: x86_64

**Configuration file:**
```
static_routes {
 0.0.0.0/0 via 10.40.20.1 dev routing table routingA
 0.0.0.0/1 via 10.40.20.3 dev routing table routingB
}

vrrp_instance nat {
  state BACKUP
  interface routing
  virtual_router_id 51
  priority 5

  advert_int 1

  authentication {
    auth_type PASS
    auth_pass BBBBBBB
  }

  unicast_src_ip 10.40.20.102

  unicast_peer {
    10.40.20.103
  }

  virtual_ipaddress {
    10.40.20.101/24 dev routing
  }


  virtual_routes {
    src 10.10.10.1 to 10.10.10.0/24 dev enp5s0 table main
  }
}

static_ipaddress {
}
```

**System Log entries**
```
Nov 30 10:05:34 nat-2 Keepalived_vrrp[33073]: Reloading
Nov 30 10:05:34 nat-2 Keepalived[33071]: Reload complete
Nov 30 10:05:34 nat-2 Keepalived[33071]: pid 33073 exited due to segmentation fault (SIGSEGV).
Nov 30 10:05:34 nat-2 Keepalived[33071]:   Please report a bug at https://github.com/acassen/keepalived/issues
Nov 30 10:05:34 nat-2 Keepalived[33071]:   and include this log from when keepalived started, a description
Nov 30 10:05:34 nat-2 Keepalived[33071]:   of what happened before the crash, your configuration file and the details below.
Nov 30 10:05:34 nat-2 Keepalived[33071]:   Also provide the output of keepalived -v, and whether keepalived is being
Nov 30 10:05:34 nat-2 Keepalived[33071]:   run in a container or VM.
Nov 30 10:05:34 nat-2 Keepalived[33071]:   A failure to provide all this information may mean the crash cannot be investigated.
Nov 30 10:05:34 nat-2 Keepalived[33071]:   If you are able to provide a stack backtrace with gdb that would really help.
Nov 30 10:05:34 nat-2 Keepalived[33071]:   Source version 2.2.7
Nov 30 10:05:34 nat-2 Keepalived[33071]:   Built with kernel headers for Linux 5.19.11
Nov 30 10:05:34 nat-2 Keepalived[33071]:   Running on Linux 6.1.0-13-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.55-1 (2023-09-29)
Nov 30 10:05:34 nat-2 Keepalived[33071]:   Command line: '/usr/sbin/keepalived' '--dont-fork'
Nov 30 10:05:34 nat-2 Keepalived[33071]:   configure options: --build=x86_64-linux-gnu --prefix=/usr --includedir=${prefix}/include
Nov 30 10:05:34 nat-2 Keepalived[33071]:                      --mandir=${prefix}/share/man --infodir=${prefix}/share/info --sysconfdir=/etc
Nov 30 10:05:34 nat-2 Keepalived[33071]:                      --localstatedir=/var --disable-option-checking --disable-silent-rules
Nov 30 10:05:34 nat-2 Keepalived[33071]:                      --libdir=${prefix}/lib/x86_64-linux-gnu --runstatedir=/run
Nov 30 10:05:34 nat-2 Keepalived[33071]:                      --disable-maintainer-mode --disable-dependency-tracking --enable-snmp --enable-sha1
Nov 30 10:05:34 nat-2 Keepalived[33071]:                      --enable-snmp-rfcv2 --enable-snmp-rfcv3 --enable-dbus --enable-json --enable-bfd
Nov 30 10:05:34 nat-2 Keepalived[33071]:                      --enable-regex --with-init=systemd build_alias=x86_64-linux-gnu CFLAGS=-g -O2
Nov 30 10:05:34 nat-2 Keepalived[33071]:                      -ffile-prefix-map=/build/keepalived-m8ENAG/keepalived-2.2.7=.
Nov 30 10:05:34 nat-2 systemd[1]: keepalived.service: Got notification message from PID 43423, but reception only permitted for main PID 33071
Nov 30 10:05:34 nat-2 Keepalived[33071]:                      -fstack-protector-strong -Wformat -Werror=format-security LDFLAGS=-Wl,-z,relro
Nov 30 10:05:34 nat-2 Keepalived[33071]:                      CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2
Nov 30 10:05:34 nat-2 Keepalived[33071]:   Config options: NFTABLES LVS REGEX VRRP VRRP_AUTH VRRP_VMAC JSON BFD OLD_CHKSUM_COMPAT SNMP_V3_FOR_V2
Nov 30 10:05:34 nat-2 Keepalived[33071]:                   SNMP_VRRP SNMP_CHECKER SNMP_RFCV2 SNMP_RFCV3 DBUS INIT=systemd SYSTEMD_NOTIFY
Nov 30 10:05:34 nat-2 Keepalived[33071]:   System options: VSYSLOG MEMFD_CREATE IPV6_MULTICAST_ALL IPV4_DEVCONF LIBNL3 RTA_ENCAP RTA_EXPIRES
Nov 30 10:05:34 nat-2 Keepalived[33071]:                   RTA_NEWDST RTA_PREF FRA_SUPPRESS_PREFIXLEN FRA_SUPPRESS_IFGROUP FRA_TUN_ID
Nov 30 10:05:34 nat-2 Keepalived[33071]:                   RTAX_CC_ALGO RTAX_QUICKACK RTEXT_FILTER_SKIP_STATS FRA_L3MDEV FRA_UID_RANGE
Nov 30 10:05:34 nat-2 Keepalived[33071]:                   RTAX_FASTOPEN_NO_COOKIE RTA_VIA FRA_PROTOCOL FRA_IP_PROTO FRA_SPORT_RANGE
Nov 30 10:05:34 nat-2 Keepalived[33071]:                   FRA_DPORT_RANGE RTA_TTL_PROPAGATE IFA_FLAGS LWTUNNEL_ENCAP_MPLS LWTUNNEL_ENCAP_ILA
Nov 30 10:05:34 nat-2 Keepalived[33071]:                   NET_LINUX_IF_H_COLLISION LIBIPVS_NETLINK IPVS_DEST_ATTR_ADDR_FAMILY
Nov 30 10:05:34 nat-2 Keepalived[33071]:                   IPVS_SYNCD_ATTRIBUTES IPVS_64BIT_STATS IPVS_TUN_TYPE IPVS_TUN_CSUM IPVS_TUN_GRE
Nov 30 10:05:34 nat-2 Keepalived[33071]:                   VRRP_IPVLAN IFLA_LINK_NETNSID GLOB_BRACE GLOB_ALTDIRFUNC INET6_ADDR_GEN_MODE VRF
Nov 30 10:05:34 nat-2 Keepalived[33071]:                   SO_MARK
Nov 30 10:05:34 nat-2 Keepalived[33071]: VRRP child process(33073) died: Respawning
Nov 30 10:05:34 nat-2 Keepalived[33071]:   Please log an issue at https://github.com/acassen/keepalived/issues/
Nov 30 10:05:34 nat-2 Keepalived[33071]:   and include a full copy of your keepalived configuration files, and
Nov 30 10:05:34 nat-2 Keepalived[33071]:   copies of the keepalived system log entries around the time this happened
```

**Did keepalived coredump?**
```
Core was generated by `/usr/sbin/keepalived --dont-fork -lm'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000556a943ea82b in ?? ()
(gdb) bt
#0  0x0000556a943ea82b in ?? ()
#1  0x0000556a943e6e1e in ?? ()
#2  0x0000556a943e7351 in ?? ()
#3  0x0000556a9444ed76 in process_threads ()
#4  0x0000556a9440d69d in start_vrrp_child ()
#5  0x0000556a943dca41 in ?? ()
#6  0x0000556a9444ed76 in process_threads ()
#7  0x0000556a943dec4d in keepalived_main ()
#8  0x00007fccccc601ca in __libc_start_call_main (
    main=main@entry=0x556a943db9a0 <main>, argc=argc@entry=3, 
    argv=argv@entry=0x7fffc4a52658)
    at ../sysdeps/nptl/libc_start_call_main.h:58
#9  0x00007fccccc60285 in __libc_start_main_impl (main=0x556a943db9a0 <main>, 
    argc=3, argv=0x7fffc4a52658, init=<optimized out>, fini=<optimized out>, 
    rtld_fini=<optimized out>, stack_end=0x7fffc4a52648)
    at ../csu/libc-start.c:360
#10 0x0000556a943db9d1 in _start ()
```

**Comment by pqarmitage** [comment]
Many thanks for reporting this. This was resolved by commit f46327dc which was applied after v2.2.7 but was included in v2.2.8.

If you upgrade to v2.2.8 it should fix the segfault.

**Comment by pqarmitage** [comment]
Investigating this issue identified another problem with static routes which is resolved by commit 707bbdf.
